### PR TITLE
Tooltip component examples

### PR DESCRIPTION
**Pull request by jakeroseman**
Two basic example and one advanced example of how to implement the tooltip material component

### CODE REVIEW

File: compose/snippets/src/main/java/com/example/compose/snippets/components/Tooltips.kt [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        positionProvider = TooltipDefaults.rememberPlainTooltipPositionProvider(),
+        tooltip = {
+            PlainTooltip { Text(plainTooltipText) }
+        },
+        state = rememberTooltipState()
+    ) {
+        IconButton(onClick = { /* Icon button's click event */ }) {
+            Icon(
+                imageVector = Icons.Filled.Favorite,
+                contentDescription = "Localized Description"
```
Comment (JolandaVerhoef) [inline compose/snippets/src/main/java/com/example/compose/snippets/components/Tooltips.kt]: What about making this `plainTooltipText` as well? "Localized Description" does not really work for blind people to understand what's going on :)

File: compose/snippets/src/main/java/com/example/compose/snippets/components/Tooltips.kt [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+                title = { Text(richTooltipSubheadText) }
+            ) {
+                Text(richTooltipText)
+            }
+        }, state = tooltipState
+    ) {
+        IconButton(onClick = { /* Icon button's click event */ }) {
+            Icon(
+                imageVector = Icons.Filled.Info,
+                contentDescription = "Localized Description"
```
Comment (JolandaVerhoef) [inline compose/snippets/src/main/java/com/example/compose/snippets/components/Tooltips.kt]: same here

File: compose/snippets/src/main/java/com/example/compose/snippets/components/Tooltips.kt [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+        tooltip = {
+            RichTooltip(
+                title = { Text(richTooltipSubheadText) },
+                action = {
+                    Row {
+                        TextButton(onClick = { /* Do something... */ }) {
+                            Text(richTooltipActionText)
+                        }
+                        TextButton(onClick = { /* Do something... */ }) {
+                            Text("Learn more")
```
Comment (JolandaVerhoef) [inline compose/snippets/src/main/java/com/example/compose/snippets/components/Tooltips.kt]: Why is one hard-coded, and the other passed as a parameter?

File: compose/snippets/src/main/java/com/example/compose/snippets/components/Tooltips.kt [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+    TooltipBox(
+        modifier = modifier,
+        positionProvider = TooltipDefaults.rememberRichTooltipPositionProvider(),
+        tooltip = {
+            RichTooltip(
+                title = { Text(richTooltipSubheadText) },
+                action = {
+                    Row {
+                        TextButton(onClick = { /* Do something... */ }) {
```
Comment (JolandaVerhoef) [inline compose/snippets/src/main/java/com/example/compose/snippets/components/Tooltips.kt]: I think we can make the dismiss action work, that would be nice

**Review by JolandaVerhoef — CHANGES_REQUESTED** [review]
There's a problem with the behavior of the rich tooltips. After I open them and then dismiss them, I can't open them again. Except when I first show another tooltip.

File: compose/snippets/src/main/java/com/example/compose/snippets/components/Tooltips.kt (hunk 3 above)
Comment (jakeroseman) [reply compose/snippets/src/main/java/com/example/compose/snippets/components/Tooltips.kt]: Well observed. I've just removed the second one.

File: compose/snippets/src/main/java/com/example/compose/snippets/components/Tooltips.kt (hunk 4 above)
Comment (jakeroseman) [reply compose/snippets/src/main/java/com/example/compose/snippets/components/Tooltips.kt]: Done.

File: compose/snippets/src/main/java/com/example/compose/snippets/components/Tooltips.kt (hunk 1 above)
Comment (jakeroseman) [reply compose/snippets/src/main/java/com/example/compose/snippets/components/Tooltips.kt]: Done.

File: compose/snippets/src/main/java/com/example/compose/snippets/components/Tooltips.kt (hunk 2 above)
Comment (jakeroseman) [reply compose/snippets/src/main/java/com/example/compose/snippets/components/Tooltips.kt]: Done.

**Review by JolandaVerhoef — APPROVED** [review]

**Pull request merged**